api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/freemarker/DeleteIndividualController.java

```diff
@@ -0,0 +1,198 @@
+
+	private static final String DEFAULT_DELETE_QUERY_TEXT = "DESCRIBE ?individualURI";
+
+	@Override
+	protected AuthorizationRequest requiredActions(VitroRequest vreq) {
+		return SimplePermission.DO_FRONT_END_EDITING.ACTION;
+	}
+
+	protected ResponseValues processRequest(VitroRequest vreq) {
+		String errorMessage = handleErrors(vreq);
+		if (!errorMessage.isEmpty()) {
+			return prepareErrorMessage(errorMessage);
+		}
+		String individualUri = vreq.getParameter("individualUri");
+		String type = getObjectMostSpecificType(individualUri, vreq);
+		Model displayModel = vreq.getDisplayModel();
+
+		String delteQueryText = getDeleteQueryForType(type, displayModel);
+		Model toRemove = getIndividualsToDelete(individualUri, delteQueryText, vreq);
+		if (toRemove.size() > 0) {
+			deleteIndividuals(toRemove, vreq);
+		}
+		String redirectUrl = getRedirectUrl(vreq);
+
+		return new RedirectResponseValues(redirectUrl, HttpServletResponse.SC_SEE_OTHER);
+	}
+
+	private String getRedirectUrl(VitroRequest vreq) {
+		String redirectUrl = vreq.getParameter("redirectUrl");
+		if (redirectUrl != null) {
+			return redirectUrl;
+		}
+		return "/";
+	}
+
+	private TemplateResponseValues prepareErrorMessage(String errorMessage) {
+		HashMap<String, Object> map = new HashMap<String, Object>();
+		map.put("errorMessage", errorMessage);
+		return new TemplateResponseValues("error-message.ftl", map);
+	}
+
+	private String handleErrors(VitroRequest vreq) {
+		String uri = vreq.getParameter("individualUri");
+		if (uri == null) {
+			return "Individual uri is null. No object to delete.";
+		}
+		if (uri.contains(">")) {
+			return "Individual uri shouldn't contain >";
+		}
+		return "";
+	}
+
+	private static String getDeleteQueryForType(String typeURI, Model displayModel) {
+		String deleteQueryText = DEFAULT_DELETE_QUERY_TEXT;
+		Query queryForTypeSpecificDeleteQuery = QueryFactory.create(queryForDeleteQuery);
+		QuerySolutionMap initialBindings = new QuerySolutionMap();
+		initialBindings.add("associatedURI", ResourceFactory.createResource(typeURI));
+		displayModel.enterCriticalSection(Lock.READ);
+		try {
+			QueryExecution qexec = QueryExecutionFactory.create(queryForTypeSpecificDeleteQuery, displayModel,
+					initialBindings);
+			try {
+				ResultSet results = qexec.execSelect();
+				if (results.hasNext()) {
+					QuerySolution solution = results.nextSolution();
+					deleteQueryText = solution.get("deleteQueryText").toString();
+				}
+			} finally {
+				qexec.close();
+			}
+		} finally {
+			displayModel.leaveCriticalSection();
+		}
+
+		if (!deleteQueryText.equals(DEFAULT_DELETE_QUERY_TEXT)) {
+			log.debug("For " + typeURI + " found delete query \n" + deleteQueryText);
+		} else {
+			log.debug("For " + typeURI + " delete query not found. Using defalut query \n" + deleteQueryText);
+		}
+		return deleteQueryText;
+	}
+
+	private String getObjectMostSpecificType(String individualURI, VitroRequest vreq) {
```

Comment:
Consider just retrieving the individual via vreq.getWebappDaoFactory() and calling getMostSpecificType() to avoid having the separate type query here.

Reply:
thanks